nautobot/extras/models/approvals.py

```diff
@@ -138,6 +141,48 @@ def __str__(self):
         """Stringify instance."""
         return f"{self.approval_workflow.name}: {self.object_under_review} ({self.current_state})"
 
+    @property
+    def current_stage(self):
+        """
+        Return the current stage of the workflow.
+        1. Return the stage that is denied if there is a stage that has been denied.
+        2. Return the first stage that is pending, if there is no stage that has been approved or denied.
+        3. Return the stage that is pending that is immediately after the last stage that has been approved.
```

Comment:
These two cases appear to be equivalent and handled by a single check below - might simplify the docstring?